### PR TITLE
using Promise.all to await for promises in array

### LINKED ISSUE #2: package is not 'exactly' like Bluebird's Promise.map
shouldn't this [line](https://github.com/yoshuawuyts/promise-map/blob/master/index.js#L14) be `return Promise.all(val.map(fn, ctx))` ? if you only do `Promise.resolve` and in the `map` function you return another promise (or promises chain) the resolved array doesn't have the final resolved values.

i have experienced this behaviour and changing `Promise.resolve` to `Promise.all` resolves the issue.

### PR DESCRIPTION
**Pull request by bjrmatos**
resolves #2

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -26,3 +26,23 @@ test('promise-map should map a fn', function(t) {
...
+    return Promise.resolve(val + 1);
+  }
+
+  function checkFn(val) {
+    t.deepEqual([3, 4, 5], val)
+  }
+
+  function handleErr() {
+    t.fail('catch')
+  }
```
Comment (yoshuawuyts) [inline test.js]: there's no need for this function; it lowers code cov and if it fails we'll know. Thanks!

Comment (bjrmatos) [inline test.js]: done! thnks for the tip :)

**Comment by bjrmatos** [comment]
coveralls not ran in my last push

but here's a screenshot with the new percentage of coverage

image

**Comment by yoshuawuyts** [comment]
published as `v1.2.0` - thanks heaps! :sparkles:

**Pull request merged**